templates/scripts/gen_template_docs.py

```diff
@@ -20,15 +20,11 @@
 GIT_REPO = "https://github.com/jboss-container-images/redhat-sso-7-openshift-image.git"
 REPO_NAME = "redhat-sso-7-openshift-image/"
 TEMPLATE_DOCS = "docs/"
-template_dirs = [ 'templates' ]
+template_dirs = [ 'templates', 'templates/openj9' ]
 amq_ssl_desc = None
 
-LINKS =  { "redhat-sso-cd-openshift:1.0": "../../templates/sso-cd-openshift{outfilesuffix}[`redhat-sso-7-tech-preview/sso-cd-openshift`]",
-           "redhat-sso-cd-openshift:5.0": "../../templates/sso-cd-openshift{outfilesuffix}[`redhat-sso-7-tech-preview/sso-cd-openshift`]",
-           "redhat-sso-cd-openshift:6.0": "../../templates/sso-cd-openshift{outfilesuffix}[`redhat-sso-7-tech-preview/sso-cd-openshift`]",
-           "redhat-sso-cd-openshift:6": "../../templates/sso-cd-openshift{outfilesuffix}[`redhat-sso-7-tech-preview/sso-cd-openshift`]",
-           "redhat-sso-cd-openshift-rhel8:7.0": "../../templates/sso-cd-openshift-rhel8{outfilesuffix}[`redhat-sso-cd-tech-preview/sso-cd-openshift-rhel8`]",
-           "redhat-sso-cd-openshift-rhel8:7": "../../templates/sso-cd-openshift-rhel8{outfilesuffix}[`redhat-sso-cd-tech-preview/sso-cd-openshift-rhel8`]"}
+LINKS =  { "redhat-sso74-openshift-rhel:1.0": "../../templates/sso74-openshift{outfilesuffix}[`redhat-sso-7/sso74-openshift`]",
```

Comment:
@mhajas The `LINKS` hash can be removed altogether (doesn't seem to be used anymore) AFAICT. Proof of concept:

```
$ git clone -b KEYCLOAK-12845 git@github.com:mhajas/redhat-sso-7-openshift-image.git
```

```
$ cd redhat-sso-7-openshift-image/templates/scripts/
```

```
$ git diff gen_template_docs.py
diff --git a/templates/scripts/gen_template_docs.py b/templates/scripts/gen_template_docs.py
index 9a1d59b..f758c85 100755
--- a/templates/scripts/gen_template_docs.py
+++ b/templates/scripts/gen_template_docs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # gen_template_doc.py
 # Kyle Liberti <kliberti@redhat.com>, Jonathan Dowland <jdowland@redhat.com>
 # ver:  Python 3
@@ -23,8 +23,8 @@ TEMPLATE_DOCS = "docs/"
 template_dirs = [ 'templates', 'templates/openj9' ]
 amq_ssl_desc = None
 
-LINKS =  { "redhat-sso74-openshift-rhel:1.0": "../../templates/sso74-openshift{outfilesuffix}[`redhat-sso-7/sso74-openshift`]",
-           "redhat-sso74-openj9-openshift-rhel:1.0": "../../templates/openj9/sso74-openj9-openshift{outfilesuffix}[`redhat-sso-7/sso74-openshift`]"}
+#LINKS =  { "redhat-sso74-openshift-rhel:1.0": "../../templates/sso74-openshift{outfilesuffix}[`redhat-sso-7/sso74-openshift`]",
+#           "redhat-sso74-openj9-openshift-rhel:1.0": "../../templates/openj9/sso74-openj9-openshift{outfilesuffix}[`redhat-sso-7/sso74-openshift`]"}
 
 PARAMETER_VALUES = {"APPLICATION_DOMAIN": "secure-app.test.router.default.local", \
                    "SOURCE_REPOSITORY_URL": "https://github.com/jboss-openshift/openshift-examples.git", \
```

```
$ ./gen_template_docs.py 
Generating docs/templates/sso74-https.adoc...
Generating docs/templates/sso74-mysql-persistent.adoc...
Generating docs/templates/sso74-mysql.adoc...
Generating docs/templates/sso74-postgresql-persistent.adoc...
Generating docs/templates/sso74-postgresql.adoc...
Generating docs/templates/sso74-x509-https.adoc...
Generating docs/templates/sso74-x509-mysql-persistent.adoc...
Generating docs/templates/sso74-x509-postgresql-persistent.adoc...
Generating docs/templates/openj9/sso74-openj9-https.adoc...
Generating docs/templates/openj9/sso74-openj9-mysql-persistent.adoc...
Generating docs/templates/openj9/sso74-openj9-mysql.adoc...
Generating docs/templates/openj9/sso74-openj9-postgresql-persistent.adoc...
Generating docs/templates/openj9/sso74-openj9-postgresql.adoc...
Generating docs/templates/openj9/sso74-openj9-x509-https.adoc...
Generating docs/templates/openj9/sso74-openj9-x509-mysql-persistent.adoc...
Generating docs/templates/openj9/sso74-openj9-x509-postgresql-persistent.adoc...
```

```
$ git status
On branch KEYCLOAK-12845
Your branch is up to date with 'origin/KEYCLOAK-12845'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   ../../docs/templates/openj9/sso74-openj9-https.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-mysql-persistent.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-mysql.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-postgresql-persistent.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-postgresql.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-x509-https.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-x509-mysql-persistent.adoc
	modified:   ../../docs/templates/openj9/sso74-openj9-x509-postgresql-persistent.adoc
	modified:   gen_template_docs.py

no changes added to commit (use "git add" and/or "git commit -a")
```
Just to retest openjdk templates would be generated properly too:
```
[iankko@localhost templates]$ pwd
/tmp/github/redhat-sso-7-openshift-image/templates
```
- Make some demo change (bump image stream tag in PostgreSQL persistent templates to `12`):
```
for template in $(ls | grep postgresql-persistent); do \
    sed -i "/\"name\": \"POSTGRESQL_IMAGE_STREAM_TAG\",/{N;s/\"value\": \"10\",/\"value\": \"12\",/g}" \
              $template; \
done
```

- Verify they got changed:
```
$ git diff *-persistent*
diff --git a/templates/sso74-postgresql-persistent.json b/templates/sso74-postgresql-persistent.json
index 7aa4f14..6e0e718 100644
--- a/templates/sso74-postgresql-persistent.json
+++ b/templates/sso74-postgresql-persistent.json
@@ -256,7 +256,7 @@
             "displayName": "PostgreSQL Image Stream Tag",
             "description": "The tag to use for the \"postgresql\" image stream. Typically, this aligns with the major.minor version of PostgreSQL.",
             "name": "POSTGRESQL_IMAGE_STREAM_TAG",
-            "value": "10",
+            "value": "12",
             "required": true
         },
         {
diff --git a/templates/sso74-x509-postgresql-persistent.json b/templates/sso74-x509-postgresql-persistent.json
index 728fdee..b61518e 100644
--- a/templates/sso74-x509-postgresql-persistent.json
+++ b/templates/sso74-x509-postgresql-persistent.json
@@ -158,7 +158,7 @@
             "displayName": "PostgreSQL Image Stream Tag",
             "description": "The tag to use for the \"postgresql\" image stream.  Typically, this aligns with the major.minor version of PostgreSQL.",
             "name": "POSTGRESQL_IMAGE_STREAM_TAG",
-            "value": "10",
+            "value": "12",
             "required": true
         },
         {
```

- Regenerate the docs:
```
cd scripts/
```

```
$ ./gen_template_docs.py
```

```
$ git diff ../../docs/templates/sso74-x509-postgresql-persistent.adoc
diff --git a/docs/templates/sso74-x509-postgresql-persistent.adoc b/docs/templates/sso74-x509-postgresql-persistent.adoc
index 08cd862..80505d3 100644
--- a/docs/templates/sso74-x509-postgresql-persistent.adoc
+++ b/docs/templates/sso74-x509-postgresql-persistent.adoc
@@ -42,7 +42,7 @@ https://docs.openshift.org/latest/architecture/core_concepts/templates.html#para
 |`SSO_REALM` | `SSO_REALM` | Realm to be created in the RH-SSO server (e.g. demorealm). | `${SSO_REALM}` | False
 |`SSO_SERVICE_USERNAME` | `SSO_SERVICE_USERNAME` | The username used to access the RH-SSO service. This is used by clients to create the appliction client(s) within the specified RH-SSO realm. | `${SSO_SERVICE_USERNAME}` | False
 |`SSO_SERVICE_PASSWORD` | `SSO_SERVICE_PASSWORD` | The password for the RH-SSO service user. | `${SSO_SERVICE_PASSWORD}` | False
-|`POSTGRESQL_IMAGE_STREAM_TAG` | -- | The tag to use for the "postgresql" image stream.  Typically, this aligns with the major.minor version of PostgreSQL. | 10 | True
+|`POSTGRESQL_IMAGE_STREAM_TAG` | -- | The tag to use for the "postgresql" image stream.  Typically, this aligns with the major.minor version of PostgreSQL. | 12 | True
 |`MEMORY_LIMIT` | -- | Container memory limit. | 1Gi | False
 |=======================================================================
```

TL; DR: So AFAICT the `LINKS` hash can be removed completely, as it doesn't seem to be used any more for the RH-SSO images.

@jmtd Hey Jon, do you know what is / was the purpose of the `LINKS` associative array in the `gen_template_docs.py` script / what it was used for?